NuRadioMC/utilities/medium_base.py

```diff
@@ -375,6 +376,192 @@ def _compute_default_ice_model_radiopropa(self):
+        up or down along the z direction. For simple models this is almost never
+        but it is used to construct more complex ice models which rely on exp.
+        profiles also
+
+        Parameters
+        ----------
+        exp_model:    pointer
+                        exponential parametrisation of the density effect on the refractive index
+        bir_model:    str
+                        choice for the birefringence model, 'southpole_A', 'southpole_B', ...
+        n_ice:  float, dimensionless
+                refractive index of the deep bulk ice
+        delta_n:  float, NuRadio length units
+                  difference between n_ice and the refractive index
+                  of the snow at the surface
+        z_0:  float, NuRadio length units
+              scale depth of the exponential
+        z_shift:  float, NuRadio length units
+                  up or down shift od the exponential profile
+        """
+
+        super().__init__(n_ice, delta_n, z_0, z_shift)
+        
+        self.exp_model = exp_model
+        self.bir_model = bir_model
+
+        self.f1 = interpolate.UnivariateSpline._from_tck(self.bir_model[0])
+        self.f2 = interpolate.UnivariateSpline._from_tck(self.bir_model[1])
+        self.f3 = interpolate.UnivariateSpline._from_tck(self.bir_model[2])
+
+        self.comp = self.exp_model.get_index_of_refraction(np.array([0, 0, -2500]))
+
+
+    def get_index_of_refraction(self, position):
+        """
+        returns the index of refraction at position.
+        Overwrites function of the mother class
+
+        Parameters
+        ----------
+        position:  1D or 2D numpy array
+                    Either one position or an array
+                    of positions for which the indices
+                    of refraction are returned
+
+        Returns
+        -------
+        n:  float
+            index of refraction
+        """
+        if isinstance(position, list) or position.ndim == 1:
+            if (position[2] - self.z_air_boundary) <= 0:
+                return self.n_ice - self.delta_n * np.exp((position[2] - self.z_shift) / self.z_0)
+            else:
+                return 1
+        else:
+            ior = self.n_ice - self.delta_n * np.exp((position[:, 2] - self.z_shift) / self.z_0)
+            ior[position[:, 2] >= 0] = 1.
+            return ior
+    
+    
+    #def get_birefringence_index_of_refraction(self, position):
+    def get_birefringence_index_of_refraction(self, position):
+        """
+        returns the index of refraction at position.
+        Overwrites function of the mother class
+
+        Parameters
+        ----------
+        position:  3dim np.array
+                    point
+
+        Returns
+        -------
+        n:  list [nx, ny, nz]
+            index of refraction for every direction
+        """
+
+        nx = self.exp_model.get_index_of_refraction(position) + self.f1(-position[2]) - self.comp
```

Comment:
some additional explanation would be good for why `self.comp` needs to be subtracted. I guess because the birefringence n(z) profile is parameterized assuming a constant density resulting in n=1.78?

Reply:
Correct, added a comment in the doc string of the `analyticalraytracer`, where this subtraction is done now.